bin/i18n/sync-in.rb

```diff
@@ -9,820 +9,821 @@
 require 'fileutils'
 require 'json'
 require 'digest/md5'
-require 'cdo/aws/metrics'
-require 'aws-sdk-ec2'
-require 'net/http'
 
 require_relative 'hoc_sync_utils'
 require_relative 'i18n_script_utils'
 require_relative 'redact_restore_utils'
 require_relative '../animation_assets/manifest_builder'
 
-I18N_METRICS_NAMESPACE = 'I18n'.freeze
-
-def sync_in
-  puts "Sync in starting"
-  Services::I18n::CurriculumSyncUtils.sync_in
-  HocSyncUtils.sync_in
-  localize_level_and_project_content
-  localize_block_content
-  localize_animation_library
-  localize_shared_functions
-  localize_course_offerings
-  localize_standards
-  localize_docs
-  puts "Copying source files"
-  I18nScriptUtils.run_bash_script "bin/i18n-codeorg/in.sh"
-  localize_external_sources
-  localize_course_resources
-  redact_level_content
-  redact_block_content
-  redact_docs
-  redact_script_and_course_content
-  redact_labs_content
-  localize_markdown_content
-  puts "Sync in completed successfully"
-rescue => exception
-  puts "Sync in failed from the error: #{exception}"
-  raise exception
-end
+module I18n
+  module SyncIn
+    module_function
```

Comment:
I think it's important to use the general ruby style guide for an open-source community-driven project, but I don't insist. I can change it if needed.
- https://github.com/rubocop/ruby-style-guide#module_function
- https://github.com/rubocop/ruby-style-guide#modules-vs-classes

Reply:
Definitely agree with that overall principle! In this situation, though, our typical use case is not explicitly addressed by the Ruby style guide. Specifically, what we usually do in this situation rather than `extend self` or `module_function` is something like:

```ruby
module SomeModule
  def self.some_method
    # body omitted
  end

  def self.some_other_method
    # body omitted
  end
end
```

My understanding (which is mostly based on [this blog post](https://www.mendelowski.com/docs/ruby/self-vs-extend-self-vs-module-function/)) is that defining methods with `self` is functionally very similar to `module_function` but with less potential for unexpected behavior when the method gets overridden.

That being said, the fact that `module_function` is explicitly encouraged by Ruby's style guide whereas our approach isn't referenced at all as either good or bad does make me think that if we were going to pick one approach to standardize around it should probably be `module_function`.

There's definitely no _need_ for a change here; what you have looks just fine! We should plan to follow this up with a consistency pass of some kind in the future, though. Thanks!